sentry-android-core/src/main/java/io/sentry/android/core/util/ConnectivityChecker.java

```diff
@@ -23,12 +24,18 @@ private ConnectivityChecker() {}
    *
    * @return true if the application has internet access
    */
-  public static @Nullable Boolean isConnected(@NotNull Context context, @NotNull ILogger logger) {
-    ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
+  public static @Nullable Boolean isConnected(
+      final @NotNull Context context, final @NotNull ILogger logger) {
+    final ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
     if (connectivityManager == null) {
       return null;
     }
-    return isConnected(context, connectivityManager, logger);
+    try {
+      return isConnected(context, connectivityManager, logger);
+    } catch (SecurityException ignored) {
+      // if no permission, we assume its connected.
```

Comment:
`m`: I would also add this comment to the doc of this method.

Reply:
added as a comment to `@return`, fwiw this is an internal class.